src/components/MemoryShardApp.tsx

```
@@ -437,28 +439,110 @@ export default function MemoryShardApp() {
+      // back oddly. Leaving the state unknown keeps the index card up, and the
+      // card already knows how to say it is still loading.
+      if (typeof status.setupComplete !== "boolean") return;
+      setState({ enabled: status.enabled === true, setupComplete: status.setupComplete });
+    } catch {
+      // Same rule: an unreachable status says nothing about setup.
+    } finally {
+      setResolved(true);
+    }
+  }, []);
+
+  useEffect(() => { void loadState(); }, [loadState]);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/components/MemoryShardApp.tsx:480:26
  478 |   }, []);
  479 |
> 480 |   useEffect(() => { void loadState(); }, [loadState]);
      |                          ^^^^^^^^^ Avoid calling setState() directly within an effect
  481 |
  482 |   const showWizard = state !== null && !state.setupComplete;
  483 |

(react-hooks/set-state-in-effect)